tests/validation/tests/v3_api/test_node_template.py

```diff
@@ -0,0 +1,246 @@
+    assert len(credentials) == 1
+    remove_resource(cloud_credential)
+
+
+@if_test_rbac
+def test_rbac_cloud_credential_list(remove_resource):
+    # User client should be able to list credential it has created
+    cloud_credential = create_cloud_credential_do(client)
+    credentials = client.list_cloud_credential(name=cloud_credential.name)
+    assert len(credentials) == 1
+    # Admin should be able to list credential
+    credentials = admin_client.list_cloud_credential(name=cloud_credential.name)
+    assert len(credentials) == 1
+    # User 2 should not be able to list credential
+    credentials2 = user1_client.list_cloud_credential(name=cloud_credential.name)
+    assert len(credentials2) == 0
+    remove_resource(cloud_credential)
+
+
+@if_test_rbac
+def test_rbac_cloud_credential_delete(remove_resource):
+    # User client should be able to delete credential it has created
+    cloud_credential = create_cloud_credential_do(client)
+    # User1 should be able to delete own credential
+    client.delete(cloud_credential)
+    credentials = client.list_cloud_credential(name=cloud_credential.name)
+    assert len(credentials) == 0
+    # Admin should be able to delete credential created by user1
+    cloud_credential2 = create_cloud_credential_do(client)
+    admin_client.delete(cloud_credential2)
+    credentials = client.list_cloud_credential(name=cloud_credential2.name)
+    assert len(credentials) == 0
+    # User 2 should not be able to delete credential created by user1
+    cloud_credential3 = create_cloud_credential_do(client)
+    with pytest.raises(ApiError) as e:
+            user1_client.delete(cloud_credential3)
+    assert e.value.error.status == 403
+    remove_resource(cloud_credential3)
+
+
+@if_test_rbac
+def test_rbac_cloud_credential_edit(remove_resource):
+    # User client should be able to edit credential it has created
+    cloud_credential = create_cloud_credential_do(client)
+    # User1 should be able to edit own credential
+    do_cloud_credential_config = {"name": "testName1"}
+    client.update(cloud_credential,
+                  digitaloceancredentialConfig=do_cloud_credential_config)
+    # Admin should be able to edit credential created by user1
+    do_cloud_credential_config = {"name": "testname2"}
+    admin_client.update(cloud_credential,
+                  digitaloceancredentialConfig=do_cloud_credential_config)
+    # User 2 should not be able to edit credential created by user1
+    with pytest.raises(ApiError) as e:
+        do_cloud_credential_config = {"name": "testname3"}
+        user1_client.update(cloud_credential,
+                            digitaloceancredentialConfig=
+                            do_cloud_credential_config)
+    assert e.value.error.status == 403
+    remove_resource(cloud_credential)
+
+
+@if_test_rbac
+def test_rbac_cloud_credential_deploy_cluster(remove_resource):
+    # Admin should be able to use credential to create cluster
```

Comment:
Could we have a more elaborate description? 
```
"""
Admin user should be able to deploy a cluster using the cloud 
credential and node template created by a standard user - User1
"""
```

Reply:
Adressed in changes.